### PR TITLE
Bump kernel version to 6.5.9

### PR DESCRIPTION
**Pull request by alexmikhalevich**
Related to https://github.com/cartesi/image-kernel/issues/15

### CODE REVIEW

File: arch/riscv/configs/defconfig [hunk 1]
```diff
@@ -240,3 +240,5 @@ CONFIG_RCU_EQS_DEBUG=y
 # CONFIG_FTRACE is not set
 # CONFIG_RUNTIME_TESTING_MENU is not set
 CONFIG_MEMTEST=y
+CONFIG_EFI=y
```
Comment (mpolitzer) [inline arch/riscv/configs/defconfig:243]: I know this was done in an old commit.
But we should revert this change and have a separate `cartesi_defconfig` instead.

I'll add an issue for this.

**Review by mpolitzer — APPROVED** [review]

File: arch/riscv/configs/defconfig (hunk 1 above)
Comment (mpolitzer) [reply arch/riscv/configs/defconfig:243]: The PR is good as is.
I think we can do this change at a later moment.

File: crypto/Kconfig [hunk 2]
```diff
@@ -1038,6 +1038,16 @@ config CRYPTO_SHA3
 	help
 	  SHA-3 secure hash algorithms (FIPS 202, ISO/IEC 10118-3)
 
+config CRYPTO_KECCAK
```
Comment (edubart) [inline crypto/Kconfig:1041]: We should also revert this after the new tools library changes.

**Review by edubart — APPROVED** [review]

File: crypto/Kconfig (hunk 2 above)
Comment (mpolitzer) [reply crypto/Kconfig:1041]: I'll do the revert here: https://github.com/cartesi/linux/pull/4

**Pull request merged**